index.js

```diff
@@ -4,27 +4,50 @@ const { inject, uninject } = require('powercord/injector');
 const { React, getModule } = require('powercord/webpack');
 const { open } = require('powercord/modal');
 
-const StringPart = require('./Components/StringPart');
+const ToneIndicator = require('./Components/ToneIndicator');
 const Modal = require('./Components/Modal');
 const Settings = require('./Components/Settings');
 
 const { getIndicator } = require('./util');
 
 const { MenuItem } = getModule(['MenuItem'], false);
 
+const LOOKBEHIND_PATTERN = /\p{P}|\s$/u;
+const INDICATOR_PATTERN = /^\/([a-z]+)(?=\p{P}|$|\s)/i;
+
 module.exports = class ToneIndicators extends Plugin {
 	async startPlugin() {
-		const parser = await getModule(['parse', 'parseTopic']);
-		const process = this.process.bind(this);
+		const parser = this.parser = await getModule(['parse', 'parseTopic']);
+
+		await this.createEmbedTitleRules();
+
+		parser.defaultRules.toneIndicator = {
+			order: parser.defaultRules.text.order - 1,
+			match: (source, state) => {
+				if (state.prevCapture && !LOOKBEHIND_PATTERN.test(state.prevCapture)) {
```

Comment:
Is `prevCapture` controlled by Discord?

Reply:
nah see https://github.com/Khan/simple-markdown#extension-overview